### PR TITLE
[Kotlin Migration] BasicAudioClipFieldController 

### PR DESCRIPTION
**Pull request by codingtosh**
[Kotlin Migration] BasicAudioClipFieldController

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicAudioClipFieldController.kt [hunk 1]
```diff
@@ -13,182 +13,178 @@
...
+import com.ichi2.anki.R
+import com.ichi2.anki.UIUtils.showThemedToast
+import com.ichi2.compat.CompatHelper
+import com.ichi2.ui.FixedTextView
+import timber.log.Timber
+import java.io.File
+
+class BasicAudioClipFieldController : FieldControllerBase(), IFieldController {
+    private var storingDirectory: File? = null
+    private var tvAudioClip: TextView? = null
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicAudioClipFieldController.kt:40]: `@KotlinCleanup("Convert to FixedTextView and make lateinit")` - allows further reduction in code

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicAudioClipFieldController.kt [hunk 2]
```diff
@@ -13,182 +13,178 @@
...
-     */
-    @VisibleForTesting
-    static String checkFileName(String audioClipFullName) {
-        return audioClipFullName.replaceAll("[^\\w.]+", "_");
+    override fun onDone() { /* nothing */
     }
 
-    @Override
-    public void onDone() { /* nothing */ }
+    override fun onFocusLost() { /* nothing */
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicAudioClipFieldController.kt:169]: This seems incorrect, should have the comment on the next line

**Review by david-allison-1 — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicAudioClipFieldController.kt (hunk 1 above)
Comment (codingtosh) [reply AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicAudioClipFieldController.kt:40]: @david-allison-1  Should I add FIELD as an annotation target in the KotlinCleanup file?

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicAudioClipFieldController.kt:40]: Yep!

**Review by david-allison-1 — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
No fault of this PR, just that component is gnarly. Thanks for the conversion!

**Pull request merged**